http-server-tck/src/main/java/io/micronaut/http/server/tck/tests/filter/RequestFilterCompletableFutureFutureProceedTest.java

```diff
@@ -0,0 +1,76 @@
+package io.micronaut.http.server.tck.tests.filter;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.annotation.*;
+import io.micronaut.http.tck.AssertionUtils;
+import io.micronaut.http.tck.HttpResponseAssertion;
+import io.micronaut.http.tck.TestScenario;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+@SuppressWarnings({
+        "java:S5960", // We're allowed assertions, as these are used in tests only
+        "checkstyle:MissingJavadocType",
+        "checkstyle:DesignForExtension"
+})
+public class RequestFilterCompletableFutureFutureProceedTest {
+    public static final String SPEC_NAME = "RequestFilterCompleteableFutureProceedTest";
+
+    @Test
+    public void requestFilterProceedWithCompletableFuture() throws IOException {
+        TestScenario.builder()
+                .specName(SPEC_NAME)
+                .request(HttpRequest.GET("/foobar").header("X-FOOBAR", "123"))
+                .assertion((server, request) -> AssertionUtils.assertDoesNotThrow(server, request, HttpResponseAssertion.builder()
+                        .status(HttpStatus.ACCEPTED)
+                        .build()))
+                .run();
+
+        TestScenario.builder()
+                .specName(SPEC_NAME)
+                .request(HttpRequest.GET("/foobar"))
+                .assertion((server, request) -> AssertionUtils.assertThrows(server, request, HttpResponseAssertion.builder()
+                        .status(HttpStatus.UNAUTHORIZED)
+                        .build()))
+                .run();
+    }
+
+    /*
+    //tag::clazz[]
+    @ServerFilter(ServerFilter.MATCH_ALL_PATTERN)
+    class CsrfFilter {
+    //end::clazz[]
+    */
+    @Requires(property = "spec.name", value = SPEC_NAME)
+    @ServerFilter(ServerFilter.MATCH_ALL_PATTERN)
+    static class FooBarFilter {
+    //tag::methods[]
+        @RequestFilter
+        @Nullable
+        public CompletableFuture<@Nullable HttpResponse<?>> filter(@NonNull HttpRequest<?> request) {
```

Comment:
Please note that if you remove the `@Nullable` annotation. It does not work. This PR ensures the users finds the correct return type incantation. He needs to use `CompletableFuture<@Nullable HttpResponse<?>>` not `CompletableFuture<HttpResponse<?>>`. 


```suggestion
        public CompletableFuture<HttpResponse<?>> filter(@NonNull HttpRequest<?> request) {
```